src/main/java/io/confluent/copycat/hdfs/HdfsWriter.java

```diff
@@ -519,4 +589,28 @@ public RecordWriter getRecordWriter(TopicPartition topicPart) {
   public WAL getWAL(TopicPartition topicPart) {
     return wals.get(topicPart);
   }
+
+  private void getVersion(TopicPartition topicPart) throws IOException {
+    String path = FileUtils.directoryName(url, topicsDir, topicPart);
+    if (compatibility.equals("NONE")) {
+      return;
+    }
+    // No files from previous execution
+    if (!offsets.containsKey(topicPart)) {
+      return;
+    }
+    long offset = offsets.get(topicPart);
+    PathFilter filter = new CommittedFileWithEndOffsetFilter(offset);
+    FileStatus[] statuses = storage.listStatus(path, filter);
+    assert statuses.length == 1;
+
+    SeekableInput input = new FsInput(statuses[0].getPath(), conf);
+    DatumReader<Object> reader = new GenericDatumReader<>();
+    FileReader<Object> fileReader = DataFileReader.openReader(input, reader);
+    org.apache.avro.Schema avroSchema = fileReader.getSchema();
+    int version = Integer.parseInt(avroSchema.getProp("version"));
+    Schema schema = avroData.toCopycatSchema(avroSchema);
+    versions.put(topicPart, version);
```

Comment:
Isn't the version in the schema anyway? I found the two maps a bit confusing at first, and now I think they may be redundant?